dev/releases/release_notes.py

```diff
@@ -165,16 +167,33 @@ def get_pr_list(date: str, extra: str) -> List[Dict[str, Any]]:
 def pr_to_md(pr: Dict[str, Any]) -> str:
     """Returns markdown string for the PR entry"""
     k = pr["number"]
-    title = pr["title"]
-    return f"- [#{k}](https://github.com/oscar-system/Oscar.jl/pull/{k}) {title}\n"
+    if has_label(pr, 'release notes: use body'):
+        mdstring = pr['body'].replace("- ", f"- [#{k}](https://github.com/oscar-system/Oscar.jl/pull/{k}) ")
+    else:
+        title = pr["title"]
+        mdstring = f"- [#{k}](https://github.com/oscar-system/Oscar.jl/pull/{k}) {title}\n"        
+    return mdstring
+
+def body_to_release_notes(pr):
+    body = pr['body']
+    index1 = body.lower().find("## release notes")
+    if index1 == -1:
+        ## not found
+        ## complain and return fallback
+        print(f"Release notes section not found in PR number {pr['number']}!!")
+        return body
+    index2 = body.find('---', index1)
+    # there are 17 characters from index 1 until the next line
+    mdstring = body[index1+17:index2]
```

Comment:
But what if there is a line `## release notes follow below` instead? Or if the trailing `---` is missing? All of these can easily happen by accident.

How about instead using a regex like this (untested, just meant as a rough guide), and then extracting the correcting matching group. (Of course multiline regex matching must be enabled for this, however one does that in Python)
```
r"^## release notes\n(.+)^---\n"
```